src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -359,7 +359,12 @@ private bool IsCountableAndIndexable(SyntaxNode node, TypeSymbol inputType, out
             }
             else
             {
-                hasErrors |= !TryBindLengthOrCount(node, receiverPlaceholder, out lengthAccess, diagnostics);
```

Comment:
Surprised that this wasn't sufficient. Pretty sure we had tests with missing members. How this was still possible?

Reply:
The tests with missing/bad length all used an `int` indexer, so produced other errors. But when we have a `System.Index` indexer, we'd produce a bound tree with an error but no diagnostics.